src/lib/token.js

```diff
@@ -1,15 +1,22 @@
 import jwt from "jsonwebtoken";
+import { APIError, ErrorTypes } from "../api/middleware/errorHandler.js";
+import { StatusCodes as HTTPStatus } from "http-status-codes/build/cjs/status-codes.js";
+import ValidationReasons from "../api/middleware/validators/validationReasons.js";
 
 export const generateToken = (data, secret, expiresInSeconds) => jwt.sign(
     { ...data },
     secret,
     { expiresIn: `${expiresInSeconds} seconds`, algorithm: "HS256" }
 );
 
-export const verifyAndDecodeToken = (token, secret) => {
+export const verifyAndDecodeToken = (token, secret, next) => {
     try {
         return jwt.verify(token, secret, { algorithm: "HS256" });
-    } catch (err) {
-        return null;
+    } catch (jwtErr) {
+        if (jwtErr.name === "TokenExpiredError") {
+            return next(new APIError(HTTPStatus.FORBIDDEN, ErrorTypes.FORBIDDEN, ValidationReasons.EXPIRED_TOKEN));
+        } else {
+            return next(new APIError(HTTPStatus.FORBIDDEN, ErrorTypes.FORBIDDEN, ValidationReasons.INVALID_TOKEN));
+        }
```

Comment:
I don't think it is necessary to differenciate between expired and invalid. I'd just stick with invalid. Also, needs to be updated from master @Naapperas

Reply:
The ideia was to make it easier to understand what went wrong, otherwise it seems like the same

Reply:
What is your reasoning behuind this request? @DoStini